include/Gaffer/Loop.inl

```diff
@@ -379,8 +392,15 @@ const ValuePlug *Loop<BaseType>::sourcePlug( const ValuePlug *output, const Cont
 	}
 	else if( ancestor == outPlugInternal() )
 	{
-		const int iterations = iterationsPlug()->getValue();
-		if( iterations > 0 && enabledPlug()->getValue() )
+		int iterations;
+		bool enabled;
+		{
+			typename LoopTraits<BaseType>::IterationsScope iterationsScope( context );
```

Comment:
I was wondering if this single scope could be moved higher, to cover everything. This would also catch the `enabledPlug()->getValue()` call on line 383.